### PR TITLE
refactor: Guarantor Screen compose migration

### LINKED ISSUE #2579: refactor: Migrate Guarantor Screens to Compose and implement Compose Navigation
**Summary:**

Migrate Guarantor Screens to Compose and implement Compose Navigation

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
Fixes #2579 
Guarantor Screen compose migration

Please Add Screenshots If there are any UI changes.
| XML Screen (before)                                                                                               |  Compose Screen (Now)                                                                                            | 
|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| https://github.com/openMF/mifos-mobile/assets/77102514/7f93f7e4-a549-4de1-85cf-2cde1d6d5723 | https://github.com/openMF/mifos-mobile/assets/77102514/fef91047-e17d-4784-ba7b-ee46a8e6f24a |

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorScreen.kt [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+fun AddGuarantorScreen(
+    uiState: GuarantorAddUiState,
+    guarantorItem: GuarantorPayload?,
+    navigateBack: () -> Unit,
+    onSubmitted: (GuarantorApplicationPayload) -> Unit
+) {
+    val context = LocalContext.current
+    val guarantorTypeOptions = rememberSaveable { mutableStateOf(listOf<GuarantorType>()) }
+
+    Scaffold(
```
Comment (AdityaKumdale) [inline app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorScreen.kt]: creating a common Scaffold would be efficient

File: app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                )
+        }
+    }
+
+    fun updateGuarantor(payload: GuarantorApplicationPayload){
+        viewModelScope.launch {
+            _guarantorState.value = GuarantorAddUiState.Loading
+            guarantorRepositoryImp.updateGuarantor(payload, loanId.value, guarantorItem.value?.id)?.catch { e ->
+                _guarantorState.value = GuarantorAddUiState.Error(e.message)
+            }?.collect {
```
Comment (AdityaKumdale) [inline app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorViewModel.kt:104]: Remove ? execpt value?

File: app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorScreen.kt [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+) {
+    val scrollState = rememberScrollState()
+    var firstName = rememberSaveable(stateSaver = TextFieldValue.Saver) { mutableStateOf(TextFieldValue( "")) }
+    var lastName = rememberSaveable(stateSaver = TextFieldValue.Saver) { mutableStateOf(TextFieldValue( "")) }
+    var city = rememberSaveable(stateSaver = TextFieldValue.Saver) { mutableStateOf(TextFieldValue( "")) }
+    var guarantorType = rememberSaveable { mutableStateOf(GuarantorType()) }
+
+    var firstNameError = rememberSaveable { mutableStateOf(false) }
+    var lastNameError = rememberSaveable { mutableStateOf(false) }
+    var guarantorTypeError = rememberSaveable { mutableStateOf(false) }
```
Comment (AdityaKumdale) [inline app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorScreen.kt:128]: These are never modified so val instead of var here.

**Review by therajanmaurya — APPROVED** [review]

File: app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorScreen.kt [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                    lastName = lastName,
+                    city = city,
+                    guarantorTypeId = guarantorType.id
+                )
+            )
+        }
+    }
+}
+
+
```
Comment (AdityaKumdale) [inline app/src/main/java/org/mifos/mobile/ui/guarantor/guarantor_add/AddGuarantorScreen.kt:250]: Previews for Loading ,error state

**Review by AdityaKumdale — CHANGES_REQUESTED** [review]

**Pull request merged**